// ds-node/lib/dsClient.js
'use strict';
const axios = require('axios');

class dsClient {
    constructor(sid, secret, options) {
        this.sid = sid;
        this.secret = secret;
        this.instance = axios.create({
            baseURL: 'https://api.dailystream.com',
            auth: {
                username: this.sid,
                password: this.secret
            }
        }) 
        if (options) {
            console.log(options);
        }
    }

    /*
    Expects:
        - CompanyId
        - ComSid
        - AgentId
        - Contract
        - Start/End
    */
    addCommunication(data) {
        return new Promise((resolve, reject) => {
            this.instance
                .post(`/v1/insertbookmark`, data)
                .then((res) => {
                    resolve(res.data);
                })
                .catch(reject);
        })
    }

    /* 
    Expects
    AgentId
    Status (Online/Offline)
    */
    setStatus(data) {
        return new Promise((resolve, reject) => {
            this.instance
                .post(`/v1/updateuser`, data)
                .then((res) => {
                    resolve(res.data);
                })
                .catch(reject);
        })
    }
}

module.exports = dsClient;
